### PR TITLE
[Tensor] add test checkpoint

### PR DESCRIPTION
**Pull request by Wesley-Jzy**
1. add test_checkpoint
2. remove trick for old ColoTensor

### CODE REVIEW

File: tests/test_tensor/test_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from torch.utils.checkpoint import checkpoint
+import torch
+from colossalai.utils import ColoInitContext
+
+
+def test_checkpoint():
+    class ModelwithCheckPoint(torch.nn.Module):
```
Comment (feifeibear) [inline tests/test_tensor/test_checkpoint.py:7]: move the model define to common. maybe some other places use it again.

File: tests/test_tensor/test_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        def __init__(self):
+            super().__init__()
+            self.proj = torch.nn.Linear(32, 32)
```
Comment (feifeibear) [inline tests/test_tensor/test_checkpoint.py:10]: use a smaller dim, make unitest faster.

File: tests/test_tensor/test_checkpoint.py (hunk 1 above)
Comment (feifeibear) [reply tests/test_tensor/test_checkpoint.py:7]: Why not use simpleNet and init is with checkpoint=True

File: tests/test_tensor/test_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            self.proj_ckpt = torch.nn.Linear(32, 2)
+
+        def forward(self, x):
+            x = self.proj(x)
+            x = checkpoint(self.proj_ckpt, x)
+            assert x.requires_grad
```
Comment (feifeibear) [inline tests/test_tensor/test_checkpoint.py:16]: I do think these assets is enought to check the grad checkpointing....

**Comment by feifeibear** [comment]
I concern the necessarity of this PR. Because it lacks an end-to-end grad/param checking for gradient checkpointing.

File: tests/test_tensor/test_checkpoint.py (hunk 1 above)
Comment (Wesley-Jzy) [reply tests/test_tensor/test_checkpoint.py:7]: I don't think it's a right usage for checkpoint. Using CheckPointModule as the last layer is wrong because our CheckPointModule uses forward as its function, so no process_output will be invoked. If it's the last layer, grad will disappear.

File: tests/test_tensor/test_checkpoint.py (hunk 2 above)
Comment (Wesley-Jzy) [reply tests/test_tensor/test_checkpoint.py:10]: Done

File: tests/test_tensor/test_checkpoint.py (hunk 1 above)
Comment (Wesley-Jzy) [reply tests/test_tensor/test_checkpoint.py:7]: In short, I mean the class uses forward() directly instead of apply(). So it's output without grad will not be post-processed by torch itself. It can only be a middle layer of a model.

Comment (Wesley-Jzy) [reply tests/test_tensor/test_checkpoint.py:7]: So would you like a new class in util or an update for simple-net?

**Comment by Wesley-Jzy** [comment]
> I concern the necessarity of this PR. Because it lacks an end-to-end grad/param checking for gradient checkpointing.

Checking the existence of grad only is because after reading the source code, I found the only useful code of checkpoint is to put fwd in no_grad() and re-run it in bwd.
But I know your conern. I will add correctness checking for it.

File: tests/test_tensor/test_checkpoint.py (hunk 1 above)
Comment (Wesley-Jzy) [reply tests/test_tensor/test_checkpoint.py:7]: I can use simple-net to do it. But current simple-net is wrong for checkpoint, I do some updates.
